src/nanotron/config/config.py

```diff
@@ -111,7 +111,7 @@ def __post_init__(self):
 class DataArgs:
     """Arguments related to the data and data files processing"""
 
-    dataset: Union[PretrainDatasetsArgs, NanosetDatasetsArgs]
+    dataset: Optional[Union[PretrainDatasetsArgs, NanosetDatasetsArgs]]
```

Comment:
why does "dataset` optional?

Reply:
Oh this is to fix other feature. Nanotron has an option to generate dummy datasets with infinite tokens. This is enabled when `dataset` is None: https://github.com/huggingface/nanotron/blob/d3db06acb2e3fe235ae512861ff64ee9fbc9ac11/run_train.py#L64. But without this Optional type in the config, we are unable to use this.